exporters/prometheus/src/exporter_utils.cc

```diff
@@ -167,6 +167,156 @@ std::string PrometheusExporterUtils::SanitizeNames(std::string name)
   return name;
 }
 
+std::regex INVALID_CHARACTERS_PATTERN("[^a-zA-Z0-9]");
+std::regex CHARACTERS_BETWEEN_BRACES_PATTERN("\\{(.*?)\\}");
+std::regex SANITIZE_LEADING_UNDERSCORES("^_+");
+std::regex SANITIZE_TRAILING_UNDERSCORES("_+$");
+std::regex SANITIZE_CONSECUTIVE_UNDERSCORES("[_]{2,}");
+
+std::string PrometheusExporterUtils::GetEquivalentPrometheusUnit(
+    const std::string &raw_metric_unit_name)
+{
+  if (raw_metric_unit_name.empty())
+  {
+    return raw_metric_unit_name;
+  }
+
+  std::string converted_metric_unit_name = RemoveUnitPortionInBraces(raw_metric_unit_name);
+  converted_metric_unit_name = ConvertRateExpressedToPrometheusUnit(converted_metric_unit_name);
+
+  return CleanUpString(GetPrometheusUnit(converted_metric_unit_name));
+}
+
+std::string PrometheusExporterUtils::GetPrometheusUnit(const std::string &unit_abbreviation)
+{
+  static std::map<std::string, std::string> units{// Time
+                                                  {"d", "days"},
+                                                  {"h", "hours"},
+                                                  {"min", "minutes"},
+                                                  {"s", "seconds"},
+                                                  {"ms", "milliseconds"},
+                                                  {"us", "microseconds"},
+                                                  {"ns", "nanoseconds"},
+                                                  // Bytes
+                                                  {"By", "bytes"},
+                                                  {"KiBy", "kibibytes"},
+                                                  {"MiBy", "mebibytes"},
+                                                  {"GiBy", "gibibytes"},
+                                                  {"TiBy", "tibibytes"},
+                                                  {"KBy", "kilobytes"},
+                                                  {"MBy", "megabytes"},
+                                                  {"GBy", "gigabytes"},
+                                                  {"TBy", "terabytes"},
+                                                  {"B", "bytes"},
+                                                  {"KB", "kilobytes"},
+                                                  {"MB", "megabytes"},
+                                                  {"GB", "gigabytes"},
+                                                  {"TB", "terabytes"},
```

Comment:
We've removed B and $ from mappings elsewhere, as B means `bel` in UCUM.  See https://github.com/open-telemetry/opentelemetry-java/pull/5719

Reply:
thanks, removed.